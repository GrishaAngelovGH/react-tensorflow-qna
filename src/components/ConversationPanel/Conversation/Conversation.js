const Conversation = ({ question, answers }) => {
  return (
    <div className="row justify-content-center mt-3">
      <div className="col-md-10 border rounded p-2">
        <h3 className="bg-primary text-white p-1 rounded">Question: {question}</h3>
        <h3 className="bg-secondary text-white p-1 rounded">
          {answers.length > 0 ? `Answer: ${answers[0]?.text}` : 'Could not find the answer for this question'}
        </h3>
      </div>
    </div>
  )
}

export default Conversation