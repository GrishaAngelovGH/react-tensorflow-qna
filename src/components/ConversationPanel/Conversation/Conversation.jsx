import Question from "./Question"
import Answer from "./Answer"

const Conversation = ({ ref, question, answer, isProcessing }) => (
  <div ref={ref} className="row justify-content-center mt-3 m-0">
    <div className="col-8 col-md-10 border rounded p-2">
      <Question content={question} />

      {
        isProcessing && (
          <div className="spinner-border text-primary" role="status">
            <span className="visually-hidden">Loading...</span>
          </div>
        )
      }
      {
        !isProcessing && (<Answer content={answer} />)
      }
    </div>
  </div>
)

export default Conversation