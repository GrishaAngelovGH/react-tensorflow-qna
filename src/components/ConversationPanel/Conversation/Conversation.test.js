import { render } from '@testing-library/react'

import Conversation from './Conversation'

test('should render Conversation component', () => {
  const view = render(<Conversation question="question" answers={[{ text: 'answer' }]} />)

  expect(view).toMatchSnapshot()
})

test('should render Conversation component without answers', () => {
  const view = render(<Conversation question="question" answers={[]} />)

  expect(view).toMatchSnapshot()
})