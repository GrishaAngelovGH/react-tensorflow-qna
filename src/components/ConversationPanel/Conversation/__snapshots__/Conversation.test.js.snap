// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Conversation component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row justify-content-center mt-3"
      >
        <div
          class="col-md-10 border rounded p-2"
        >
          <h3
            class="bg-primary text-white p-1 rounded"
          >
            Question: 
            question
          </h3>
          <h3
            class="bg-secondary text-white p-1 rounded"
          >
            Answer: answer
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row justify-content-center mt-3"
    >
      <div
        class="col-md-10 border rounded p-2"
      >
        <h3
          class="bg-primary text-white p-1 rounded"
        >
          Question: 
          question
        </h3>
        <h3
          class="bg-secondary text-white p-1 rounded"
        >
          Answer: answer
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Conversation component without answers 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row justify-content-center mt-3"
      >
        <div
          class="col-md-10 border rounded p-2"
        >
          <h3
            class="bg-primary text-white p-1 rounded"
          >
            Question: 
            question
          </h3>
          <h3
            class="bg-secondary text-white p-1 rounded"
          >
            Could not find the answer for this question
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row justify-content-center mt-3"
    >
      <div
        class="col-md-10 border rounded p-2"
      >
        <h3
          class="bg-primary text-white p-1 rounded"
        >
          Question: 
          question
        </h3>
        <h3
          class="bg-secondary text-white p-1 rounded"
        >
          Could not find the answer for this question
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
